packages/react-vapor/src/components/datePicker/tests/DatePickerDropdown.spec.tsx

```diff
@@ -104,14 +104,27 @@ describe('Date picker', () => {
             expect(datePickerDropdown.find('DatePickerBox').length).toBe(1);
         });
 
-        it('should only display an input if it has the readonly prop', () => {
+        it('should disable the dropdown button when readonly props is truthy', () => {
             datePickerDropdown.setProps({
                 ...DATE_PICKER_DROPDOWN_BASIC_PROPS,
                 readonly: true,
             });
+            const expectedElement = (
+                <button className="dropdown-toggle btn inline-flex flex-center dropdown-toggle-placeholder" disabled>
+                    <span className="dropdown-selected-value">
+                        <label>Select date</label>
+                    </span>
+                    <span className="dropdown-toggle-arrow"></span>
+                </button>
+            );
 
-            expect(datePickerDropdown.find('.date-picker-dropdown').length).toBe(0);
-            expect(datePickerDropdown.find('Input').length).toBe(1);
+            expect(datePickerDropdownInstance.props.readonly).toBeTruthy();
+            expect(
+                datePickerDropdown
+                    .children()
+                    .find('button.dropdown-toggle')
+                    .matchesElement(expectedElement)

```

Comment:
Instead of having to define the expectedElement and aserting `.matchesElement`, we could simply check that it has the `disabled` prop set to true.